docs/tools/efi.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+# Copyright (c) 2022 Aditya Garg
+
+set -euo pipefail
+
+if [[ `uname -s` != "Linux" ]] ; then
+	echo "This script is intended to be run from Linux. Aborting!"
+	exit 1
+fi
+
+
+if [[ ${1-default} == default ]] ; then
+	echo "Please specify the name of the new EFI partition. Aborting!"
+	exit 1
+fi
+
+if [ $USER != root ]
+then
+echo "This script is intented to be run as root. You will be asked for your password."
+sudo chmod 755 $0
+sudo $0 "$1"
+exit 0
+fi
+
+echo "Are you sure you want to change your EFI partition to \""$1"\"? (y/N)"
+read input
+if [[ ($input != y) && ($input != Y) ]]
+then
+	echo "Abort!"
+	exit 1
+fi
+
+echo "Changing the EFI partition."
+# Mount the new partition to /tmp/newefi
+
+mkdir /tmp/newefi
```

Comment:
maybe confirm that this isn't already there or there's also `mktemp` for making random temporary directories

Reply:
mktemp makes files and folders with random names so there won't be the issue of if a user has files in /tmp/newefi those files will be deleted.

There's also the option of checking that /tmp/newefi isn't already there.

Reply:
actually does mkdir fail if tge directory is already there? if it does then this wouldn't be an issue

Reply:
Yes it fails

Reply:
I don't think mktemp is required then